lib/util/propTypes.js

```diff
@@ -143,10 +143,10 @@ module.exports = function propTypesInstructions(context, components, utils) {
         }
       });
 
-      // nested object type spread means we need to ignore/accept everything in this object
-      if (containsObjectTypeSpread) {
-        return {};
-      }
+      // Mark if this shape has spread. We will know to consider all props from this shape as having propTypes,
```

Comment:
Similar to the case with propTypes, if the shape contains a spread plus some properties e.g.
```js
type Props = {
  person: {
    ...$Exact<BasePerson>,
    lastname: string
  }
};
```
it would be unwise to throw away the information we have about `lastname` just because we have a spread. By setting an additional flag and leaving the children data in place, we can detect if `lastname` is being used.